client/src/pages/Balance/Checkout.jsx

```diff
@@ -0,0 +1,232 @@
+  <div className="ErrorMessage" role="alert">
+    <svg width="16" height="16" viewBox="0 0 17 17">
+      <path
+        fill="#FFF"
+        d="M8.5,17 C3.80557963,17 0,13.1944204 0,8.5 C0,3.80557963 3.80557963,0 8.5,0 C13.1944204,0 17,3.80557963 17,8.5 C17,13.1944204 13.1944204,17 8.5,17 Z"
+      />
+      <path
+        fill="#6772e5"
+        d="M8.5,7.29791847 L6.12604076,4.92395924 C5.79409512,4.59201359 5.25590488,4.59201359 4.92395924,4.92395924 C4.59201359,5.25590488 4.59201359,5.79409512 4.92395924,6.12604076 L7.29791847,8.5 L4.92395924,10.8739592 C4.59201359,11.2059049 4.59201359,11.7440951 4.92395924,12.0760408 C5.25590488,12.4079864 5.79409512,12.4079864 6.12604076,12.0760408 L8.5,9.70208153 L10.8739592,12.0760408 C11.2059049,12.4079864 11.7440951,12.4079864 12.0760408,12.0760408 C12.4079864,11.7440951 12.4079864,11.2059049 12.0760408,10.8739592 L9.70208153,8.5 L12.0760408,6.12604076 C12.4079864,5.79409512 12.4079864,5.25590488 12.0760408,4.92395924 C11.7440951,4.59201359 11.2059049,4.59201359 10.8739592,4.92395924 L8.5,7.29791847 L8.5,7.29791847 Z"
+      />
+    </svg>
+    {children}
+  </div>
+);
+
+const stripePromise = loadStripe("pk_test_Y2tTju8cgfUZW4vozmYVG26J00yEwRHw7v");
+
+const Checkout = ({
+  refillAmount,
+  setRefillAmount,
+  setCheckoutPage,
+}) => {
+  const classes = useStyles();
+  const { user } = useContext(UserContext);
+  const stripe = useStripe();
+  const elements = useElements();
+  const [error, setError] = useState(null);
+  const [cardComplete, setCardComplete] = useState(false);
+  const [processing, setProcessing] = useState(false);
+  const [billingDetails, setBillingDetails] = useState({ name: "" });
+  const [paymentSuccessful, setPaymentSuccessful] = useState(false);
+
+  const handleSubmit = async event => {
+    event.preventDefault();
+
+    // Stripe.js has not yet loaded.
+    if (!stripe || !elements) return;
+    if (!billingDetails.name) {
+      document.getElementById("name").focus();
+      setError({ message: "Please enter your name" });
+      return;
+    }
+    if (error || !cardComplete ) {
+      elements.getElement("card").focus();
+      setError({ message: "Invalid card number" });
+      return;
+    }
+    
+    setProcessing(true);
+
+    const requestPaymentIntent = await axios.post(`/api/users/${user.id}/purchase`, { refillAmount });
+
+    const payload = await stripe.createPaymentMethod({
+      type: "card",
+      card: elements.getElement(CardElement),
+      billing_details: billingDetails,
+    });
+
+    const confirm = await stripe.confirmCardPayment(
+      `${requestPaymentIntent.data.clientSecret}`,
+      {
+        payment_method: payload.paymentMethod.id
+      }
+    );
+
+    if (confirm.paymentIntent.status === "succeeded") {
+      setProcessing(false);
+      setPaymentSuccessful(true);
+      user.balance += refillAmount;
+      setRefillAmount(1);
+      await axios.put(`/api/users/${user.id}/add-credits`, { refillAmount });
```

Comment:
as a reminder for the future, there is a security issue with this, since if people discover this API, they can make calls to it without doing a payment. There are two ways to solve this:
1. use a webhook instead to handle success (https://stripe.com/docs/payments/accept-a-payment#web-fulfillment)
2. pass the client secret to this API call and check if the payment intent was successful also on teh backend (https://stripe.com/docs/api/payment_intents/retrieve)

Reply:
Thanks! Will implement on a future PR